### ISSUE TITLE
Pulse align_measures implicitly assumes u3 gate.

### ISSUE THREAD

**Issue by nkanazawa1989** [Qiskit/qiskit-terra#5472]
### What is the current behavior?

`align_measures` function can take arbitrary `cal_gate` to decide minimum alignment time, but actually we can only take `u3` due to its implementation.

https://github.com/Qiskit/qiskit-terra/blob/fe458633cfc969a4ac4b5073396a30160b5cdaa8/qiskit/pulse/transforms.py#L30-L36

This is nested function to get the gate duration of input `cal_gate`. As you can see this tries to bind three parameters to generate a schedule of `x` gate with `u3` instruction.

https://github.com/Qiskit/qiskit-terra/blob/fe458633cfc969a4ac4b5073396a30160b5cdaa8/qiskit/pulse/transforms.py#L118-L124

If we specify `x` in `cal_gate`, this binds three parameters though `x` instruction has no parameter.

### What is the expected behavior?

This time should be a required duration to generate |1> and |0> state. However this seems to be requirements from our backend and should not be here because Qiskit is backend agnostic. This measurement discriminator cal should be handled by backend and we should be able to remove this from frontend. l think user cannot understand why this minimum duration requirement is here.

### Suggested solutions

I think I have two approaches:
1) Keep minimum duration and update schedule generation:
Add `cal_gate_parameters: List[float]` to function argument. We need to check parameter number matching with specified instruction. But we can also remove `instmap` and `cal_gate` from argument to simplify the interface.

2) Remove minimum duration requirements:
User can set arbitrary acquisition time here and the backend should realign acquisition time if they need to insert measurement cal.

**Comment by taalexander** [comment]
Another option would be to create a circuit that applies an `x` across all qubits and schedule. From that `x` extract the duration. 

Due to the way the imports are structured in the `compiler` module, this can lead to circular dependencies between `circuit -> compiler -> pulse -> compiler` but it is possible to negotiate as shown in `pulse.builder`.

**Comment by paniash** [comment]
Can I take this up? I'm quite new but would love to contribute and would appreciate some pointers.

**Comment by nkanazawa1989** [comment]
Yes, thank you for taking this issue. I like the scheduling approach because it seems to be robust to the backend change so basically you can take that approach :) You need to create circuit that applies x gate to all measured qubits without measurement and feed that circuit to the transpiler and scheduler then you can get the system's minimum alignment time. Or you can use scheduled circuit, i.e. transpling with `scheduling_method` that gives you only schedule duration without converting into pulses. In this approach you can avoid circular import.

However I'm not really sure about if we really need to take care about minimum calibration duration on frontend. I think this really depends on how the discriminator is calibrated within the job. any thought? @taalexander

**Comment by Dinesh-byte-cloud** [comment]
hi I would like to contribute to qiskit community and am very new to this. I  would  love to work on this problem and get  some pointers to solve them

**Comment by 1ucian0** [comment]
hi @paniash ! Are you still working on this one?

**Comment by paniash** [comment]
@1ucian0 Hi! Sorry about the delay. I would love to continue working on this but if it's urgent, then please feel free to reassign it to someone else.

**Comment by a-freakish** [comment]
I am new to quantum community and open source. I would love to work on this issue.

**Comment by paniash** [comment]
@a-freakish Hi! My exams just got over and I'm currently working on this. :-)

**Comment by taalexander** [comment]
Excellent! Good luck @paniash 😄 , please let us know if you require any help.

**Comment by kdk** [comment]
Hi @paniash , any update on this?

**Comment by paniash** [comment]
@kdk Hi! Sorry, life's gotten in the way. Please feel free to take over or assign this to someone else.

**Comment by rickyzcode** [comment]
@1ucian0 Hi, I am new to open source. But I am interested in this issue, can you please assign this to me? Thank you.

**Comment by javabster** [comment]
@rickyzcode sorry for the slow response, I've assigned to you now 😄

Let us know if you have any questions, and if you're new to open source I recommend taking a look at the following resources as well:
- [Contributing Guidelines](https://github.com/Qiskit/qiskit-terra/blob/main/CONTRIBUTING.md)
- [GitHub / CI tips for Qiskit Contributors](https://www.youtube.com/watch?v=-eLeOeF_-5o)

**Comment by javabster** [comment]
update: issue has been unassigned and is available for another contributor to pick up

**Comment by Yagami1729** [comment]
@javabster I would like to work on this. Can you please assign me.

**Comment by Tarun-Kumar07** [comment]
Hi @Yagami1729 , I was going through good first issues to contribute to and this issue caught my interest. If you are not actively working on it, may I take it up?

**Comment by Tarun-Kumar07** [comment]
Hi @javabster and @1ucian0,

I hope you're well. This issue I'm interested in was previously assigned to @Yagami1729. I've asked in the [comment](https://github.com/Qiskit/qiskit/issues/5472#issuecomment-1973264329) above, but haven't received a reply. If the issue is not actively being worked on, could one of you please reassign it to me?

Thanks!

**Comment by 1ucian0** [comment]
`qiskit.pulse` was deprecated in 1.3 and removed in 2.0. It has not been supported since 1.4 reached end-of-life in April 2026. Closing this issue as `not planned`.
